framework/src/main/java/org/checkerframework/common/value/qual/MatchesRegex.java

```diff
@@ -0,0 +1,28 @@
+package org.checkerframework.common.value.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.SubtypeOf;
+
+/**
+ * An annotation indicating the possible values for a String type. If an expression's type has this
+ * annotation, then at run time, the expression evaluates to a string that matches at least one of
```

Comment:
Rather than adding examples here, I've changed this to refer to the documentation of `java.lang.String#matches`. That absolves us of the need to maintain examples here, where they might become out of date. I will add the tests you describe, though, since those will start to fail if the behavior of `matches` changes.